core/src/fgm/fgj/gamejamgame/GameEvent.java

```diff
@@ -11,10 +11,13 @@
 public class GameEvent {
 	final EventContext eventContext;
 	String eventText;
-	final String eventType;
 	boolean didLose;
 	boolean isPositive;
 
+	private SolarSystem solarSystem;
+	private Planet planet;
+	private Ship ship;
+
```

Comment:
On one hand I feel like the solar system, planet, and ship aren't part of the event itself, they're just used to generate one. Having them as fields for the class itself is dishonest to the architecture and what a game event represents since the references are changed through time and a game's reference doesn't reflect the state of those resources used to make the event in the first place. On the other hand I can see having the solar system, planet, and ship that generated the event available would be handy for undoing or detailed event logs.

Reply:
This also leaves stray references to them which can manipulate game state in unexpected ways if a non-private method were to be introduced. These sorts of bugs can be avoided with diligent programming practices (like making sure everything is private) so it isn't a big deal unless we need to loosen access on the API for another feature. I'm also not sure if leaving the game vulnerable to cheaters is an issue or not. By default in a practice project or tutorial, I think these vulnerabilities should be avoided so they can be identified in real systems.